ha_lineairdb.h

```diff
@@ -274,9 +276,20 @@ class ha_lineairdb : public handler {
  private:
   std::string get_current_key();
   void set_current_key(const uchar* key = nullptr);
-  void set_write_buffer();
+
+  std::array<char, BYTE_BIT_NUMBER> interpret_buf_flag(uchar flag);
+  void set_write_buffer(uchar* buf);
   bool is_primary_key_exists();
   int is_primary_key_type_int();
+
+  bool store_blob_to_field(Field** field);
+  void flush_null_flag_to_buf(uchar* buf, std::bitset<BYTE_BIT_NUMBER> &nullBit, 
+                              int &field_index, int &buf_nullbyte_index);
+  bool is_over_buf_flag_capacity(int field_index);
+  bool set_flag_for_nonnull_field(std::bitset<BYTE_BIT_NUMBER> &nullBit, int &clm_cnt);
+  bool handle_null_field(uchar* buf, Field** field,
```

Comment:
Is it better to make them non-member functions?

Reply:
Yes, I agree. Since the member function in the header should be an interface that can be used by other classes, it is better not to make them member functions in this case.
I think we should define them not as member functions but as simple utility functions inside the source file.

However, as a general rule, if a function is called only once in the entire code, there is no need to define it as a function at all. Just write it directly in the caller.
It is painful for other coders to read this code with too many jumps.

Of course, it is important to split up long functions for readability. Thus I recommend refactoring the original long function at first. Then you would split the original long function into the main logic and utility functions.

Reply:
What do you mean by refactoring the original long function in this case?